pkg/plugin/interfaces.go

```diff
@@ -43,77 +43,84 @@ type Deprecated interface {
 	DeprecationWarning() string
 }
 
-// GenericSubcommand is an interface that defines the plugins operations
-type GenericSubcommand interface {
-	// UpdateContext updates a Context with command-specific help text, like description and examples.
+// Subcommand is an interface that defines the common base for subcommands returned by plugins
+type Subcommand interface {
+	// UpdateContext updates a Context with subcommand-specific help text, like description and examples. It also serves
+	// to pass context from the CLI to the subcommand, such as the command name.
 	// Can be a no-op if default help text is desired.
 	UpdateContext(*Context)
-	// BindFlags binds the plugin's flags to the CLI. This allows each plugin to define its own
-	// command line flags for the kubebuilder subcommand.
+	// BindFlags binds the subcommand's flags to the CLI. This allows each subcommand to define its own
+	// command line flags.
 	BindFlags(*pflag.FlagSet)
 	// Run runs the subcommand.
 	Run() error
-	// InjectConfig passes a config to a plugin. The plugin may modify the
-	// config. Initializing, loading, and saving the config is managed by the
-	// cli package.
+	// InjectConfig passes a config to a plugin. The plugin may modify the config.
+	// Initializing, loading, and saving the config is managed by the cli package.
 	InjectConfig(*config.Config)
 }
 
-// Context is the runtime context for a plugin.
+// Context is the runtime context for a subcommand.
 type Context struct {
-	// CommandName sets the command name for a plugin.
+	// CommandName sets the command name for a subcommand.
 	CommandName string
 	// Description is a description of what this subcommand does. It is used to display help.
 	Description string
-	// Examples are one or more examples of the command-line usage
-	// of this plugin's project subcommand support. It is used to display help.
+	// Examples are one or more examples of the command-line usage of this subcommand. It is used to display help.
 	Examples string
 }
 
-// InitPluginGetter is an interface that defines gets an Init plugin
-type InitPluginGetter interface {
-	Base
-	// GetInitPlugin returns the underlying Init interface.
-	GetInitPlugin() Init
+// InitPlugin is an interface for plugins that provide an `init` subcommand
+type InitPlugin interface {
```

Comment:
To avoid stuttering with the package name, how about
```suggestion
type Init interface {
```
Ditto elsewhere